src/Build/BackEnd/Components/Logging/EventSourceSink.cs

```diff
@@ -224,57 +224,59 @@ public void Consume(BuildEventArgs buildEvent, int sinkId)
         /// </summary>
         public void Consume(BuildEventArgs buildEvent)
         {
+
             switch (buildEvent)
             {
                 case BuildMessageEventArgs buildMessageEvent:
-                    RaiseMessageEvent(null, buildMessageEvent);
+                    RaiseEvent(null, buildMessageEvent, (o, args) => MessageRaised?.Invoke(o, args), (o, args) => AnyEventRaised?.Invoke(o, args));
                     break;
                 case TaskStartedEventArgs taskStartedEvent:
-                    RaiseTaskStartedEvent(null, taskStartedEvent);
+                    RaiseEvent(null, taskStartedEvent, (o, args) => TaskStarted?.Invoke(o, args), (o, args) => StatusEventRaised?.Invoke(o, args));
```

Comment:
same for StatusEventRaised as a follow up. in RaisedTaskStartedEvent is RaiseStatusEvent - StatusEventRaised with try catch

Reply:
good catch.
So if we need the try-catch, it would be this invocation instead:
  ``` RaiseEvent(buildMessageEvent, (args) => MessageRaised?.Invoke(null, args), (args) => RaiseEvent(args, (args) => MessageRaised?.Invoke(null, args), null)); ```
?
Alternatively, since this is starting to look quite lambda-ish and not readable enough, I can keep around the RaiseStatusEvent and pass it instead of the lambda since it suddenly doesn't look all that nice.

That being said, if we already succesfully passed specific event handler, shoudln't the generic one be fine as-is? (could be unnecessarily dangerous)